### PR TITLE
add stub for contextvars.pyi

### PR DESCRIPTION
**Pull request by JelleZijlstra**
See PEP 567 and https://docs.python.org/dev/library/contextvars.html#module-contextvars.

Part of #1965.

### CODE REVIEW

**Comment by gvanrossum** [comment]
@1st1 Can you have a quick look here?

File: stdlib/3.7/contextvars.pyi [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from typing import Any, Callable, ClassVar, Generic, Mapping, TypeVar, Union
+
+_T = TypeVar('_T')
+
+class ContextVar(Generic[_T]):
+    def __init__(self, name: str, *, default: _T) -> None: ...
```
Comment (1st1) [inline stdlib/3.7/contextvars.pyi]: `default` is not a required parameter.  It's not `typing.Optional` either, as it is either of type `_T` or not passed at all (`None` is not a valid value).  I'm not sure how to express this in terms of `typing`.

A couple of examples:

```python
# valid:
int_var: ContextVar[int] = ContextVar(default=1)
int_var: ContextVar[int] = ContextVar()

# invalid:
int_var: ContextVar[int] = ContextVar(default=None)
```

and

```python
# valid:
int_var: ContextVar[Optional[int]] = ContextVar(default=1)
int_var: ContextVar[Optional[int]] = ContextVar(default=None)
int_var: ContextVar[Optional[int]] = ContextVar()
```

Here's a pure Python implementaion of PEP 567 / `ContextVar`: https://github.com/MagicStack/contextvars/blob/master/contextvars/__init__.py#L111

Comment (JelleZijlstra) [reply stdlib/3.7/contextvars.pyi]: We can write `*, default: _T = ...`. Sorry for missing this.

Comment (gvanrossum) [reply stdlib/3.7/contextvars.pyi]: Jelle: If `= ...` doesn't work, you can use an overload (once without, once with `default`).

Comment (1st1) [reply stdlib/3.7/contextvars.pyi]: NP!

**Comment by 1st1** [comment]
LGTM.  

I wish there was a way to express `Token.old_value` more clearly, but since `Token.MISSING` is not a type we can't use a union type there, right?

**Comment by JelleZijlstra** [comment]
Yes, I couldn't think of a clean way. An approach I thought of was to define a fake type like `class _Missing: ...` and then define `MISSING` as being of type `_Missing`, and have `old_value` return `Union[_T, _Missing]`. However, I assume the idiomatic way of using `old_value` would be something like:
```
val = token.old_value()
if val is token.MISSING:
     # do something
else:
     # do something else
```
and with the proposed type annotations I don't think mypy would be able to infer that val is not MISSING in the else branch.

So unless Guido can think of something better, we should probably go with Any.

**Comment by 1st1** [comment]
> However, I assume the idiomatic way of using old_value would be something like:

FWIW I think that the primary use-case for `Token.old_value` would be debugging.

**Pull request merged**